engine/src/main/java/org/terasology/logic/players/LocalPlayer.java

```diff
@@ -29,16 +29,17 @@
 import org.terasology.network.ClientComponent;
 import org.terasology.physics.HitResult;
 import org.terasology.physics.Physics;
+import org.terasology.recording.EntityIdMap;
 import org.terasology.registry.CoreRegistry;
 
-/**
- */
 public class LocalPlayer {
 
     private EntityRef clientEntity = EntityRef.NULL;
     private int nextActivationId;
+    private EntityIdMap entityIdMap;
 
-    public LocalPlayer() {
+    public LocalPlayer(EntityIdMap entityIdMap) {
```

Comment:
Rather than *change* the constructor to require an `EntityIdMap` when it seems optional (one place creating a `LocalPlayer` just passes in a `new EntityIdMap()`) instead create an additional constructor and have the old one call it `this(new EntityIdMap());` - then you don't need to change existing usages that rely on the default behavior. This otherwise breaks stuff.